src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java

```diff
@@ -74,65 +84,48 @@
-      httpCon = createConnection(account, url, contentType, metadata);
-      mOutputStream  = httpCon.getOutputStream();
-      LOG.debug("Second attempt PUT {} successfull. Got output stream", url.toString());
-      mHttpCon = httpCon;
-    } catch (IOException e) {
-      LOG.error(e.getMessage());
-      throw e;
-    }
-  }
-
-  /**
-   * Creates HTTP Connection
-   *
-   * @param account JOSS Account
-   * @param url URL to the object
-   * @param contentType object content type
-   * @param metadata user provided meta data
-   * @return HttpURLConnection if success
-   * @throws IOException if error
-   */
-  private HttpURLConnection createConnection(JossAccount account, URL url, String contentType,
-      Map<String, String> metadata) throws IOException {
-    LOG.debug("Create chunked HTTP connection for PUT {}. Read timeout {}. Streaming chunk {}",
-        url.toString(), READ_TIMEOUT, STREAMING_CHUNK);
-    HttpURLConnection newHttpCon = (HttpURLConnection) url.openConnection();
-    newHttpCon.setDoOutput(true);
-    newHttpCon.setRequestMethod("PUT");
-
-    newHttpCon.addRequestProperty("X-Auth-Token",account.getAuthToken());
-    newHttpCon.addRequestProperty("Content-Type", contentType);
+    client = connectionManager.createHttpConnection();
+    request = new HttpPut(mUrl.toString());
+    request.addHeader("X-Auth-Token", account.getAuthToken());
     if (metadata != null && !metadata.isEmpty()) {
       for (Map.Entry<String, String> entry : metadata.entrySet()) {
-        newHttpCon.addRequestProperty("X-Object-Meta-" + entry.getKey(), entry.getValue());
+        request.addHeader("X-Object-Meta-" + entry.getKey(), entry.getValue());
+      }
+    }
+
+    PipedOutputStream out = new PipedOutputStream();
+    final PipedInputStream in = new PipedInputStream();
+    out.connect(in);
+    mOutputStream = out;
+    writeThread = new Thread() {
+      public void run() {
+        InputStreamEntity entity = new InputStreamEntity(in, -1);
+        entity.setChunked(true);
+        entity.setContentType(contentType);
+        request.setEntity(entity);
+        try {
+          HttpResponse response = client.execute(request);
+          int responseCode = response.getStatusLine().getStatusCode();
+          if (responseCode >= 400) {
+            LOG.warn("Http Error Code: {} for {}, \nReason:", responseCode, mUrl.toString(),
+                    response.getStatusLine().getReasonPhrase());
+          }
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
       }
+    };
+  }
+
+  private void startThread() {
+    if (writeThread.getState().equals(Thread.State.NEW)) {
+      writeThread.start();
     }
-    newHttpCon.setDoInput(true);
-    newHttpCon.setRequestProperty("Connection", "close");
-    newHttpCon.setReadTimeout(READ_TIMEOUT);
-    newHttpCon.setRequestProperty("Transfer-Encoding","chunked");
-    newHttpCon.setDoOutput(true);
-    newHttpCon.setRequestProperty("Expect", "100-continue");
```

Comment:
@djalova New code doesn't use "Expect", "100-continue". This is crucial stuff, since it prevents upload entire object if token expire. Old code used 100-continue and then catched (ProtocolException e) if token expired or not enough permissions. Exception handling then performed token retry.
We need this stuff with the new code as well.